### PR TITLE
feat(medusa): Release staging version on push to `develop`

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Modify staging action to deploy staging versions on pushes to develop using the `staging` tag

**How**
- snapshot vs pre mode

I have chosen to not use the `pre`-mode that changesets provide and rather opt for the use of a snapshot release. I have chosen to do so because as far as I can tell a snapshot release fits our requirements, however it's a simpler procedure that relies more on changesets built-in features. Also using the `pre`-mode does come with a lot of warnings not to mess things up, whereas when using snapshot releases (as we already do), is much safer provided we always release using a tag.

- Not using the `changesets/action` github action

The `changesets/action` github action creates a pr that needs to can be, however when deploying a versioned staging version we dont want any package semver upgrades created from the `version` command to be preserved. To entirely prevent this the github action instead manually runs the process of deploying using changeset.

**Testing/Demo**
I have duplicated all medusa packages scoped to my npmjs account `@pkorsholm/` for testing. In my repo I have the action from this pr running on `pull_request` and it will deploy a staging package everytime a pr is changed. Feel free to test it out by creating a pr from from the master branch, removing endpoints or whatever (it just needs to build), and make a pr. All packages with changesets will be re-deployed under the tag `staging`. In your server you need to do a couple of steps because of how we run the medusa server: 
1. install packages with staging tag
  1.a.  all packages are prefixed scoped with `@pkorsholm`, if they were previously scoped to `@medusajs` they are now also scoped to `@pkorsholm`. 
2. In order to properly find the loaders duplicate the `@pkorsholm` folder in `./node_modules`
3. rename the folder `@pkorsholm - copy` to `@medusajs` (this is where the loaders will be loaded from
4. move packages that are not `medusa` and `medusa-cli` from the newly created `@medusajs` folder into `node_modules`
5. run you server

repeat 1-5 after release of a new package under the staging tag (only required because i renamed all packages)

**Question**
- Should we remove the old staging packages when deploying a new one in order to not clog our versions on npmjs? 
  - npmjs has an api we could use

Fixes CORE-747

### CODE REVIEW

File: .github/workflows/staging-release.yml [hunk 1]
```diff
@@ -15,17 +15,32 @@ jobs:
...
+        run: npm config set scripts-prepend-node-path auto
 
       - name: Install Dependencies
         run: yarn
 
-      - name: Create Release Pull Request
-        uses: changesets/action@v1
-        with: 
-          version: yarn version:staging
+      - name: Install Dependencies build
```
Comment (olivermrbl) [inline .github/workflows/staging-release.yml:37]: todo: Rename action step
```suggestion
      - name: Build all packages
```

File: .github/workflows/staging-release.yml [hunk 2]
```diff
@@ -15,17 +15,32 @@ jobs:
...
+        run: yarn build
+
+      - name: version staging packages
```
Comment (olivermrbl) [inline .github/workflows/staging-release.yml:40]: comment: This is very nit, but just want to stay consistent 😅 
```suggestion
      - name: Version staging packages
```

File: .github/workflows/staging-release.yml [hunk 3]
```diff
@@ -15,17 +15,32 @@ jobs:
...
+        run: yarn version:staging
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: publish staging packages
```
Comment (olivermrbl) [inline .github/workflows/staging-release.yml:45]: comment: Same here 😄 
```suggestion
      - name: Publish staging packages
```

File: .github/workflows/staging-release.yml [hunk 4]
```diff
@@ -15,17 +15,32 @@ jobs:
...
+        run: yarn release:staging
```
Comment (olivermrbl) [inline .github/workflows/staging-release.yml:46]: thought: I believe we must run an install after versioning to ensure the lockfile is refreshed with the latest staging versions. I've encountered issues around this when running manual releases.

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Super nice @pKorsholm! Have left a couple of comments.

**Comment by olivermrbl** [comment]
> Should we remove the old staging packages when deploying a new one in order to not clog our versions on npmjs?
npmjs has an api we could use

I think this should be left for a follow-up later on. Will open an issue for it 👍

**Comment by pKorsholm** [comment]
Closing this for now in order to include ticket mentioned by @olivermrbl

**Comment by pKorsholm** [comment]
Re-opening this after initial investigation into unpublishing previous versions. 

The initial conclusion is that it's not possible to unpublish a given version of a package if another package depends on _any_ version of the package. Future investigations will be into publishing packages under a different organisation intended for staging only, but will be postponed until after an initial version of the staging environment is functional.

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**